sdk/cosmos/azure-cosmos/azure/cosmos/_execution_context/aio/hybrid_search_aggregator.py

```
@@ -147,21 +147,33 @@ async def _run_hybrid_search(self):
             self._format_final_results(drained_results)
             return
 
+        # Get the Components weights if any
+        if self._hybrid_search_query_info.get('componentWeights'):
+            component_weights = self._hybrid_search_query_info['componentWeights']
+            component_count = len(self._hybrid_search_query_info['componentQueryInfos'])
```

Where do we verify mismatching counts of number of components and number of weights?

---

we don't - the service is responsible for that

---

it should raise an error - and yes, along with the tests Bryan wrote for that we should also document it in the README @bambriz